src/VisualStudio/Core/Def/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -932,17 +932,15 @@ public bool Equals(DiagnosticData item1, DiagnosticData item2)
                     item1.ProjectId != item2.ProjectId ||
                     item1.Severity != item2.Severity ||
                     item1.Message != item2.Message ||
-                    item1.DataLocation.MappedStartLine != item2.DataLocation.MappedStartLine ||
-                    item1.DataLocation.MappedStartColumn != item2.DataLocation.MappedStartColumn ||
-                    item1.DataLocation.OriginalStartLine != item2.DataLocation.OriginalStartLine ||
-                    item1.DataLocation.OriginalStartColumn != item2.DataLocation.OriginalStartColumn)
+                    item1.DataLocation.MappedFileSpan?.Span != item2.DataLocation.MappedFileSpan?.Span ||
+                    item2.DataLocation.OriginalFileSpan.Span != item2.DataLocation.OriginalFileSpan.Span)
                 {
                     return false;
                 }
 
-                return (item1.DocumentId != null) ?
-                    item1.DocumentId == item2.DocumentId :
-                    item1.DataLocation.OriginalFilePath == item2.DataLocation.OriginalFilePath;
+                return (item1.DocumentId != null)
+                    ? item1.DocumentId == item2.DocumentId
+                    : item1.DataLocation.OriginalFileSpan.Path == item2.DataLocation.OriginalFileSpan.Path;
```

Comment:
why does this only look at the path, and not hte mapped path?  i have no idea.  we do look at hte mapped span above, so this is strangely inconsistent.  Note: i'm also not sure if we could ever have mapped spns here as this is the ExternalError reporter... which i would not think would ever give us mapped spans...

all this code makes me sad.